client/src/document/hooks.ts

```diff
@@ -8,3 +11,72 @@ export function useDocumentURL() {
   // pathname with the document's mdn_url.
   return url.endsWith("/") ? url.substring(0, url.length - 1) : url;
 }
+
+export function useCopyExamplesToClipboard(doc: Doc | undefined) {
+  React.useEffect(() => {
+    if (!doc) {
+      return;
+    }
+    if (!navigator.clipboard) {
+      console.log(
+        "Copy-to-clipboard disabled because your browser does not appear to support it."
+      );
+      return;
+    }
+
+    [...document.querySelectorAll("div.code-example pre")].forEach(
+      (element) => {
+        const userMessage = document.createElement("span");
+        const button = document.createElement("button");
+        const span = document.createElement("span");
+        userMessage.textContent = "Copied!";
+        span.textContent = "Copy to Clipboard";
+
+        const wrapper = element.parentElement;
+
+        userMessage.setAttribute("class", "user-message");
+        userMessage.setAttribute("aria-hidden", "true");
+        button.setAttribute("aria-hidden", "false");
+        button.setAttribute("type", "button");
+        button.setAttribute("class", "copy-icon");
+        span.setAttribute("class", "visually-hidden");
+
+        button.appendChild(span);
+        wrapper?.appendChild(button);
+        wrapper?.appendChild(userMessage);
+
+        button.onclick = async () => {
+          let copiedSuccessfully = true;
+          try {
+            const text = element.textContent || "";
+            await navigator.clipboard.writeText(text);
+          } catch (err) {
+            console.error(
+              "Error when trying to use navigator.clipboard.writeText()",
+              err
+            );
+            copiedSuccessfully = false;
+          }
+
+          if (copiedSuccessfully) {
+            userMessage.classList.add("show");
+            userMessage.setAttribute("aria-hidden", "false");
+            button.classList.add("copied");
+            userMessage.style.top = "52px";
+          } else {
+            button.classList.add("failed");
+            userMessage.classList.add("show");
+            userMessage.setAttribute("aria-hidden", "false");
+            userMessage.style.top = "52px";
+            userMessage.textContent = "Error trying to copy to clipboard!";
+          }
+
+          setTimeout(() => {
+            userMessage.classList.remove("show");
+            userMessage.setAttribute("aria-hidden", "true");
+          }, 1000);
```

Comment:
```suggestion
          }, copiedSuccessfully ? 1000 : 3000);
```
Because now that the text is longer and less common/familiar, it surprised me (when I was mocking an error) that it disappeared so quickly.

Reply:
Yeah agreed, when I was mocking the error a little bit ago, it seemed to disappear a bit to quickly.